google/cloud/storage/internal/policy_document_request.h

```diff
@@ -104,7 +104,8 @@ std::ostream& operator<<(std::ostream& os, PolicyDocumentRequest const& r);
 
 class PolicyDocumentV4Request {
  public:
-  PolicyDocumentV4Request() : scheme_("https") {}
+  PolicyDocumentV4Request()
+      : scheme_("https"), endpoint_authority_("storage.googleapis.com") {}
```

Comment:
Consider:

```suggestion
  PolicyDocumentV4Request()
      : PolicyDocumentV4("storage.googleapis.com") {}
  explicit PolicyDocumentV4Request(std::string endpoint_authority)
      : scheme_("https"), endpoint_authority_(std::move(endpoint_authority)) {}
  PolicyDocumentV4Request(PolicyDocumentV4 document, std::string endpoint_authority)
      : PolicyDocumentV4Request(std::move(endpoint_authority)),
        document_(std::move(document)) {}
```

and you can remove the `SetEndpointAuthority()` function. You may need to change a few places where the constructor from `PolicyDocumentV4` is called.